packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions-tasks/test/emr/integ.emr-create-cluster-add-step-runtime-role.ts

```diff
@@ -0,0 +1,130 @@
+import * as sfn from 'aws-cdk-lib/aws-stepfunctions';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as emr from 'aws-cdk-lib/aws-emr';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as tasks from 'aws-cdk-lib/aws-stepfunctions-tasks';
+import { App, Stack } from 'aws-cdk-lib';
+import { IntegTest } from '@aws-cdk/integ-tests-alpha';
+
+/*
+ * Create a state machine with an EMR cluster and adds a step that uses a runtime role.
+ *
+ * Stack verification steps:
+ * The generated State Machine can be executed from the CLI (or Step Functions console)
+ * and runs with an execution status of `Succeeded`.
+ *
+ * -- aws stepfunctions start-execution --state-machine-arn <state-machine-arn-from-output> provides execution arn
+ * -- aws stepfunctions describe-execution --execution-arn <state-machine-arn-from-output> returns a status of `Succeeded`
+ */
+
+const app = new App();
+const stack = new Stack(app, 'aws-cdk-emr-add-step-runtime-role');
+
+const vpc = new ec2.Vpc(stack, 'Vpc', { restrictDefaultSecurityGroup: false });
+
+const cfnSecurityConfiguration = new emr.CfnSecurityConfiguration(stack, 'EmrSecurityConfiguration', {
+  name: 'AddStepRuntimeRoleSecConfig',
+  securityConfiguration: JSON.parse(`
+    {
+      "AuthorizationConfiguration": {
+          "IAMConfiguration": {
+              "EnableApplicationScopedIAMRole": true,
+              "ApplicationScopedIAMRoleConfiguration": 
+                  {
+                      "PropagateSourceIdentity": true
+                  }
+          },
+          "LakeFormationConfiguration": {
+              "AuthorizedSessionTagValue": "Amazon EMR"
+          }
+      }
+    }`),
+});
```

Comment:
@lpizzinidev I added this as well so no prereqs are required. Step function succeeds after deploying stack as-is. Good shout.

Reply:
Nice 👍